NCGO2024/Assets/App/Resource/Scripts/UI/LobbyManager.cs

```diff
@@ -0,0 +1,225 @@
+
+    private void LeaveBtnClick()
+    {
+        if (!IsServer)
+        {
+
+            QuitLobbyServerRpc();
+
+        }
+        else
+        {
+            foreach (PlayerInfoData playerdata in _networkPlayers._allConnectedPlayers)
+            {
+
+                if (playerdata._clientID != _myLocalClientID)
+                {
+
+
+                    KickUserBtn(playerdata._clientID);
+
+                }
+            }
+
+            NetworkManager.Shutdown();
+            SceneManager.LoadScene(0);
+        }
+    }
+
+
+    [Rpc(target:SendTo.Server, RequireOwnership = false)]
+    private void QuitLobbyServerRpc(RpcParams rpcParams=default)
+    {
+        KickUserBtn(rpcParams.Receive.SenderClientId);
+    }
+
+    private void NetPlayersChanged(NetworkListEvent<PlayerInfoData> changeEvent)
+    {
+
+        Debug.Log(message: "Net Players has changed event fired");
+        PopulateLabels();
+    }
+
+    [ContextMenu(itemName:"PopulateLabel")]
+
+    private void PopulateLabels()
+    {
+        ClearPlayerPanel();
+
+
+
+
+        bool allReady = true;
+
+        foreach (PlayerInfoData playerData in _networkPlayers._allConnectedPlayers)
+        {
+            GameObject newPlayerPanel = Instantiate(_panelPrefab, _ContentGO.transform);
+
+            PlayerLabel _playerLabel = newPlayerPanel.GetComponent<PlayerLabel>();
+
+            _playerLabel.onKickClicked += KickUserBtn;
+
+            if (IsServer && playerData._clientID != _myLocalClientID) 
+            {               
+                _playerLabel.setKickActive(isOn: true);
+              
+            }
+            else
+            {
+                _playerLabel.setKickActive(isOn: false);
+            }
+
+            _playerLabel.SetPlayerLabelName(playerData._clientID);
+            _playerLabel.setReady(playerData._isPlayerReady);
+            _playerLabel.setPlayerColor(playerData._colorID);
+            _PlayerPanels.Add(newPlayerPanel);
+
+            if(playerData._isPlayerReady == false)
+            {
+                allReady = false;   
+            }
+        }
+
+        if (IsServer) 
+        {
+            if (allReady)
+            {
+                if (_networkPlayers._allConnectedPlayers.Count > 1)
+                {
+                    _rdyTxt.text = "Ready to start";
+                    _startBtn.gameObject.SetActive(true);  
+                }
+                else
+                {
+                    _rdyTxt.text = "Empty Lobby";
+                }
+
+            }
+            else
+            {
+                _rdyTxt.text = "Waiting for Ready Players";
+            }
+
+        }
+    }
+
+    private void KickUserBtn(ulong kickTarget)
+    {
+        if(!IsServer || !IsHost) return;
+
+        foreach (PlayerInfoData playerData in _networkPlayers._allConnectedPlayers)
+        {
+            if (playerData._clientID == kickTarget)
+            {
+
+                //_networkPlayers._allConnectedPlayers.Remove(playerData);
```

Comment:
uncomment remove function at kick in the LobbyManager or o finish this implementation of the  connection event on the NetworkedPlayerData